### ISSUE TITLE
Login with WordPress.com account on atomic site when commenting does not work

### ISSUE THREAD

**Issue by holdercp** [Automattic/jetpack#40665]
When trying to comment on an atomic site that requires an account to login, I cannot login with a newly created WordPress.com account.

Here's a screen capture of the flow:

https://github.com/user-attachments/assets/5215e697-af96-4eff-b211-2b86f3c6ffca

## Steps to Reproduce

1. Configure an atomic site to require authenticated users to comment: Settings -> Discussion -> Comments -> "Users must be registered and logged in to comment" checked
2. Also check "Let visitors use a WordPress.com or Facebook account to comment."

<img width="1323" alt="Image" src="https://github.com/user-attachments/assets/79642389-9598-457e-87f3-62265575581d" />

4. Visit a post on the site as an unauthenticated user and attempt to comment on the post
5. On the WordPress.com login screen, create a new account with an unused email
6. Verify the email using the link sent to your email
7. Attempt to login with the new WordPress.com account when prompted on the original login screen

## Expected

I would expect to be authenticated with this site and able to comment.

## Actual

I am presented with the WPAdmin "Account not found" errors when logging in.

## Related

This issue was also reported here: p1697634622150289-slack-C029GN3KD

**Comment by jeherve** [comment]
Transferring this issue to the Jetpack repo, where the Comments feature is developed. This repo is only used for reports about the Bugomattic tool itself.

**Comment by jeherve** [comment]
Thanks for the report! What happens once your account has been created, verified, you're logged in, and if you go back to the site and attempt to comment from scratch again? Do you experience the same issue when trying to comment?

**Comment by holdercp** [comment]
@jeherve Thanks for moving this issue to the correct repo. I must've made a mistake when reporting through Bugomattic.

Here's what happens if, after I authenticate with WPCom, go and try to leave a comment:

https://github.com/user-attachments/assets/8018b4f5-6207-4bfa-8819-0acb53ae22ea

Same issue. It seems like the connection between the WPCom account and the WP account on this instance is not being setup. Those forms and the error make it seems like it's trying to log me in as a user of the site and the WPCom account is not used.

**Comment by jeherve** [comment]
Unfortunately I still cannot seem to reproduce on my end. I wonder if this may be browser or browser extension-related. Do you experience the same issue regardless of the browser?

**Comment by holdercp** [comment]
Hmm. I tried from FF with no extensions (other than Bitwarden). I'm still experiencing an issue but it's a different one. I'm unproxied and authenticated with wp.com, which can clearly be seen in the video. When I try to comment I get a "Must be logged in" window:

https://github.com/user-attachments/assets/420e8833-f620-4404-99ae-2d42b6f79705

**Comment by jeherve** [comment]
> When I try to comment I get a "Must be logged in" window:

What are the site's discussion settings? Does it require folks to be logged in to leave a comment?

**Comment by holdercp** [comment]
Yes, here's the discussion settings:

![Image](https://github.com/user-attachments/assets/7f0db4f8-8587-43fe-a504-349f1dd313da)

What's odd is I'm authenticated with wp.com in that video above but I'm unable to leave a comment.

**Comment by jeherve** [comment]
> What's odd is I'm authenticated with wp.com in that video above but I'm unable to leave a comment.

You're authenticated with WordPress.com, but most likely not with the site. That's why you cannot comment. I would say that is the expected behavior (we wouldn't want anyone with a WordPress.com account to be able to comment on a site where the admin only wants folks with an account to be able to comment), but the user experience is definitely not great here. 

I'm not sure what the best UX would be here. I'd be curious to have others' opinions on this.

**Comment by holdercp** [comment]
What is the intent of the first setting marked in the settings screenshot?

> Let visitors use a WordPress.com account or Facebook account to comment

As an admin, I would expect someone could comment when logged in with their wp.com account with that setting turned on.

**Comment by jeherve** [comment]
Yeah, I can see how this could be confusing, you're right. The first setting effectively replaces the default comment form by a form with alternative authentication options.
Realistically, I don't think that setting should be there when you require folks to log in to your site to leave a comment.

cc @Automattic/vertex who has been working on improving the commenting experience.

**Comment by holdercp** [comment]
Ah, I think I understand. You're saying it might be related to conflicting settings around comments? The top setting uses a different mechanism for authenticating and commenting, while the bottom setting is the standard wp auth'ed comment setting. Only one of those should really be active at a time: comment via logging in with social/wp.com _or_ comment as a user of the site.

**Comment by jeherve** [comment]
Yup, exactly that!

**Comment by escapemanuele** [comment]
As we (Vertex) are not following Verbum anymore, I'd like to tag @arcangelini 👍

**Comment by arcangelini** [comment]
Added to the Loop board. Will take a look when I get some time.

**Comment by davemart-in** [comment]
Based on the comments, I'm closing this out as expected behavior. Happy to re-open it if customers write in complaining about similar confusion.
